### ISSUE TITLE
SDK 43 , You need to add `ACCESS_BACKGROUND_LOCATION` to the AndroidManifest.]

### ISSUE THREAD

**Issue by islamouzou** [expo/expo#14774]
### Summary

whenever i try to start background location i get this warning , it was fine on sdk 42

You need to add `ACCESS_BACKGROUND_LOCATION` to the AndroidManifest.]

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

43

### Environment

 Expo CLI 4.12.1 environment info:
    System:
      OS: Windows 10 10.0.19042
    Binaries:
      Node: 14.16.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.10 - C:\Users\lenovo\AppData\Roaming\npm\yarn.CMD
      npm: 6.14.12 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7583922
    npmPackages:
      expo: ^43.0.0-beta.4 => 43.0.0-beta.4
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.2 => 0.64.2
      react-native-web: 0.17.1 => 0.17.1
    Expo Workflow: managed

**Comment by brentvatne** [comment]
please share a reproducible example in order to complete the issue template correctly. https://stackoverflow.com/help/minimal-reproducible-example

**Comment by islamouzou** [comment]
`       let { status } = await Location.getBackgroundPermissionsAsync();

        if (status == "granted") {
          await Location.hasStartedLocationUpdatesAsync("locationupdate").then(
            async function (result) {
            
              if (user && user.user && user.user.connected) {
               

                let { status } =
                  await Location.requestBackgroundPermissionsAsync();
                if (status == "granted") {
                  startLocation();
                }
              }
            }
          );
        } else {
          setShowAlert(true);
        }
      })();
    }


const startLocation = () => {
    Location.startLocationUpdatesAsync("locationupdate", {
      accuracy: Location.Accuracy.BestForNavigation,
      timeInterval: 10000,
      distanceInterval: 300,
      showsBackgroundLocationIndicator: true,
      foregroundService: {
        notificationTitle: "En ligne ... ",
        notificationBody: "Mise à jour de votre position en cours ...",
      },
      pausesUpdatesAutomatically: false,
    });`

**Comment by islamouzou** [comment]
let { status } = await Location.getBackgroundPermissionsAsync();

        if (status == "granted") {
          await Location.hasStartedLocationUpdatesAsync("locationupdate").then(
            async function (result) {
            
              if (user && user.user && user.user.connected) {
               

                let { status } =
                  await Location.requestBackgroundPermissionsAsync();
                if (status == "granted") {
                  startLocation();
                }
              }
            }
          );
        } else {
          setShowAlert(true);
        }
      })();
    }


      const startLocation = () => {
    Location.startLocationUpdatesAsync("locationupdate", {
      accuracy: Location.Accuracy.BestForNavigation,
      timeInterval: 10000,
      distanceInterval: 300,
      showsBackgroundLocationIndicator: true,
      foregroundService: {
        notificationTitle: "En ligne ... ",
        notificationBody: "Mise à jour de votre position en cours ...",
      },
      pausesUpdatesAutomatically: false,
    });

**Comment by islamouzou** [comment]
On the SDK 42 it's working correctly  .

**Comment by islamouzou** [comment]
@byCedric  thanks for your reply , i tried now to create a new project for a minimal example  and i had the same warning , this time i didn't use forground service . this happens only on expo client .


`  const startLocation = () => {
    Location.startLocationUpdatesAsync("locationupdate", {
      accuracy: Location.Accuracy.Highest,
      timeInterval: 15000,
      distanceInterval: 1,
      showsBackgroundLocationIndicator: true,
      pausesUpdatesAutomatically: false,
    });
  };`

**Comment by ayman-rahmon** [comment]
i have the same problem and i'm using : 

  Expo CLI 4.12.1 environment info:
    System:
      OS: Linux 5.14 Arch Linux
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.11.1 - /usr/bin/node
      npm: 8.0.0 - /usr/bin/npm
    npmPackages:
      expo: ~42.0.1 => 42.0.3 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.12.1
    Expo Workflow: managed
    
    
    
    
i have written down all of the  details in my question on stackoverflow : 


https://stackoverflow.com/questions/69786649/react-native-expo-app-access-background-location-needs-to-be-added-to-the-manife    
    


Note : i do request the foreground permission but i don't use it to track the location of the device so it shouldn't be the problem i think . based on the answer provided by @byCedric on this thread.

i also just tried asking for foreground before background , asking for only background and all nothing works and i keep getting the same output

**Comment by kefaya-app** [comment]
> @byCedric according to the documentation my snack should work: https://snack.expo.dev/@helpling/backgroundlocation
> 
> but on expo `2.22.3` it's not (while it is on `2.21.5`). Do you see something wrong there?
> 
> Btw. I closed my bug #14937 because I thought it was a duplicate of this one, but I'm not sure anymore.

i now have tried on the older version and it's as you said . 

it's working on the older version with no problems

**Comment by ayman-rahmon** [comment]
> @byCedric according to the documentation my snack should work: https://snack.expo.dev/@helpling/backgroundlocation
> 
> but on expo `2.22.3` it's not (while it is on `2.21.5`). Do you see something wrong there?
> 
> Btw. I closed my bug #14937 because I thought it was a duplicate of this one, but I'm not sure anymore.

 i just tried on the older version of expo go and it seems to be working now .... so yeah i think this was the problem all along

**Comment by mRcSchwering** [comment]
Same issue here.

```
  Expo CLI 4.12.11 environment info:
    System:
      OS: Linux 5.11 Ubuntu 20.04.2 LTS (Focal Fossa)
      Shell: 5.0.17 - /bin/bash
    Binaries:
      Node: 16.9.0 - /usr/local/bin/node
      Yarn: 1.22.15 - /usr/bin/yarn
      npm: 7.21.1 - /usr/local/bin/npm
    npmPackages:
      expo: ~43.0.0 => 43.0.1 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.2 => 0.64.2 
      react-native-web: 0.17.1 => 0.17.1 
    Expo Workflow: managed
```

I can't use version `2.21.5`: "This project requires a newer version of Expo Go"

**Comment by bracketized** [comment]
Same issue here. I've tried many things:

- Removing permissions on app.json or adding all of them known
- Downgrading expo client ( asks for newer version as for SDK43 ).
- Changing packages versions from expo-location.
- Using old expo-permissions package.

No luck so far, all of them throws the same manifest issue regarding location.
I need to publish and now google is throwing my builds because i must show a video
demostrating background location useage, but i can't since it's not working.

It was on SDK42, but i do not pretend to downgrade after that much work done.

I've also opened an issue linked to here to keep track of changes:
https://forums.expo.dev/t/weird-error-with-background-location-permissions/58239

**Comment by FelipeACP** [comment]
> Same issue here. I've tried many things:
> 
> * Removing permissions on app.json or adding all of them known
> * Downgrading expo client ( asks for newer version as for SDK43 ).
> * Changing packages versions from expo-location.
> * Using old expo-permissions package.
> 
> No luck so far, all of them throws the same manifest issue regarding location. I need to publish and now google is throwing my builds because i must show a video demostrating background location useage, but i can't since it's not working.
> 
> It was on SDK42, but i do not pretend to downgrade after that much work done.
> 
> I've also opened an issue linked to here to keep track of changes: https://forums.expo.dev/t/weird-error-with-background-location-permissions/58239

The easiest way would be to build and install the apk directly (without play store) so that you can record your demonstration since the issue lies on the expo client.
Downgrading expo client will only work for SDK 42 or older, unfortunately.

**Comment by bracketized** [comment]
> 
> 
> > Same issue here. I've tried many things:
> > 
> > * Removing permissions on app.json or adding all of them known
> > * Downgrading expo client ( asks for newer version as for SDK43 ).
> > * Changing packages versions from expo-location.
> > * Using old expo-permissions package.
> > 
> > No luck so far, all of them throws the same manifest issue regarding location. I need to publish and now google is throwing my builds because i must show a video demostrating background location useage, but i can't since it's not working.
> > It was on SDK42, but i do not pretend to downgrade after that much work done.
> > I've also opened an issue linked to here to keep track of changes: https://forums.expo.dev/t/weird-error-with-background-location-permissions/58239
> 
> The easiest way would be to build and install the apk directly (without play store) so that you can record your demonstration since the issue lies on the expo client. Downgrading expo client will only work for SDK 42 or older, unfortunately.

Well, the thing is that even with this i'm unable to use the background location permission, and i'm unable to debbug it properly, quite weird all of it, but i will retry again tonight.

Thanks!

**Comment by brentvatne** [comment]
hey all! you can work around this by using expo-dev-client if you like: https://docs.expo.dev/clients/introduction/

it seems like we may need to add this permission back to expo go. we'll follow up on this next week

**Comment by bracketized** [comment]
> 
> 
> hey all! you can work around this by using expo-dev-client if you like: https://docs.expo.dev/clients/introduction/
> 
> it seems like we may need to add this permission back to expo go. we'll follow up on this next week

I'm unable to get up and running the custom client, i will keep trying solutions to make it work but i would prefer to just use the expo go withouth anything else.

Thanks

**Comment by ReidyPlayerOne** [comment]
Any updates on this?

**Comment by khelifioussama** [comment]
Always the same problem 
You need to add `ACCESS_BACKGROUND_LOCATION` to the AndroidManifest.
i added permissions: ["ACCESS_BACKGROUND_LOCATION"] underr android in app.config.js
I am requesting foreground permission before requestion background permissions and it didn't work 
Any updates please ?

**Comment by gate5th** [comment]
Has anyone had any success with this? Especially with Android 11? I've been struggling with getting background location permissions to work for almost 2 months:

- tried every combination of properties for startLocationUpdatesAsync
- have background permissions selected on all the time
- have it running in a custom dev client that has ACCESS_BACKGROUND_LOCATION permission in app.json
- I request foreground permissions, do a location check, and then request background permissions afterward as detailed here: https://developer.android.com/training/location/permissions

The only thing I've gotten to work (just yesterday) is foreground permissions using Expo43 and custom dev client. Unfortunately that's very awkward since my use case is a driving monitoring app so it needs to be on all the time (or anytime there is significant movement at least) 

Can anyone recommend how to find help with this? I know other apps must do it somehow 🤕 I'd happily pay a consultant at this point

**Comment by FelipeACP** [comment]
> Has anyone had any success with this? Especially with Android 11? I've been struggling with getting background location permissions to work for almost 2 months:
> 
> * tried every combination of properties for startLocationUpdatesAsync
> * have background permissions selected on all the time
> * have it running in a custom dev client that has ACCESS_BACKGROUND_LOCATION permission in app.json
> * I request foreground permissions, do a location check, and then request background permissions afterward as detailed here: https://developer.android.com/training/location/permissions
> 
> The only thing I've gotten to work (just yesterday) is foreground permissions using Expo43 and custom dev client. Unfortunately that's very awkward since my use case is a driving monitoring app so it needs to be on all the time (or anytime there is significant movement at least)
> 
> Can anyone recommend how to find help with this? I know other apps must do it somehow 🤕 I'd happily pay a consultant at this point
You want to add ACCESS_BACKGROUND_LOCATION to your dev client manifest, see [Config Plugins](https://docs.expo.dev/guides/config-plugins/) or [Customizing Your Runtime](https://docs.expo.dev/clients/getting-started/#customizing-your-runtime) for possible ways.

**Comment by linderlee17** [comment]
Alguien con alguna solucion, tengo exactamente el mismo problema

**Comment by brentvatne** [comment]
@FelipeACP - also, the `expo-location` plugin can be configured to add this permission: https://github.com/expo/expo/blob/master/packages/expo-location/plugin/src/withLocation.ts

**app.json**:
```json
{
  "expo": {
    // ..
    "plugins": [
      ["expo-location", { isAndroidBackgroundLocationEnabled: true } ]
    ]
  }
}
```

do a [development build](https://docs.expo.dev/build/eas-json/#development-builds) with this config and you will be able to use background location

**Comment by DeluxeOwl** [comment]
@brentvatne Any update on this?

**Comment by gate5th** [comment]
@brentvatne @FelipeACP I appreciate the help, but still no dice. I have added the below to app.json, done dev clients as well as a "preview" build using the EAS build and still no background tracking.  Does anyone have example project that is working for background location tracking (on Android 11 or 12) that could be a template?

```
"plugins": [
            ["expo-location", { "isAndroidBackgroundLocationEnabled": true }]
        ],
```

and under "android"
```
"permissions": [
                "ACCESS_COARSE_LOCATION",
                "ACCESS_FINE_LOCATION",
                "ACCESS_BACKGROUND_LOCATION"
            ],
```

**Comment by assidiwassim** [comment]
Any updates on this ?

**Comment by eimenanacreado** [comment]
Any update on this ?

**Comment by eimenanacreado** [comment]
The solution i had found is to back to SDK 42, here's steps to do that !

-  First of all download the 2.21.5 Expo Apk ( For Android from [here ](https://d1ahtucjixef4r.cloudfront.net/Exponent-2.21.5.apk) )
-  Into packages.json change the version of your dependencies ( You can pick up my json or the version of what you want )
- Run npm install
- Run expo install
- On Android Play store disable auto update for expo go application, to prevent that to update to a new version
- npm start -- --reset-cache

`"dependencies": {
    "@react-native-async-storage/async-storage": "~1.15.0",
    "@react-navigation/native": "^6.0.6",
    "@react-navigation/native-stack": "^6.2.5",
    "axios": "^0.24.0",
    "expo": "^42.0.5",
    "expo-background-fetch": "~9.2.2",
    "expo-constants": "~11.0.1",
    "expo-font": "~9.2.1",
    "expo-linear-gradient": "~9.2.0",
    "expo-location": "~12.1.2",
    "expo-status-bar": "~1.0.4",
    "expo-task-manager": "~9.2.2",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-native": "0.63.4",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-reanimated": "^2.2.4",
    "react-native-safe-area-context": "3.2.0",
    "react-native-screens": "~3.4.0",
    "react-native-web": "0.17.1",
    "react-redux": "^7.2.6",
    "redux-persist": "^6.0.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@types/react": "~17.0.21",
    "@types/react-native": "~0.64.12",
    "typescript": "~4.3.5"
  },`

Little change between expo 42 and 34 on Location & Background Task here's what you need to put : 

For Location background : 

`const askLocationPermission = async () => {

        await Location.requestForegroundPermissionsAsync();
        let { status } = await Location.requestBackgroundPermissionsAsync();

        if (status !== 'granted') {
            handleLocationError('Permission to access location was denied');
            console.debug('Permission to access location was denied')
        } else {
            handleAllowedLocation(true);
            await startLocationTracking();
        }

    }

    const startLocationTracking = async () => {

        await Location.startLocationUpdatesAsync(BACKGROUND_FETCH_TASK, {
            accuracy: Location.Accuracy.BestForNavigation,
            timeInterval: 5000,
            foregroundService: {
                notificationTitle: "LOCATION_TITLE",
                notificationBody: "LOCATION_SUBTITLE",
                notificationColor: "red"
            }
        });
    }`

For Background Task : 

TaskManager.defineTask(BACKGROUND_FETCH_TASK, async ({ data, error }) => {

    if (error) {
        return;
    }

     console.debug('Received new locations : ');
     console.debug(data);


    return BackgroundFetch.Result.NewData;
});

Hope the problem with SDK 43 Will resolved soon ♥

**Comment by endronk** [comment]
Right! There are two things here:
- If you want to try on Expo, you need to use [Expo v2.21.5](https://d1ahtucjixef4r.cloudfront.net/Exponent-2.21.5.apk) because the latest one has issue (see [this](https://github.com/expo/expo/issues/14774#issuecomment-952947261)).
- If you want to use a standalone app (APK), you need to add `ACCESS_BACKGROUND_LOCATION ` permission under `android`
```
"permissions": [
    "ACCESS_BACKGROUND_LOCATION",
    "ACCESS_COARSE_LOCATION",
    "ACCESS_FINE_LOCATION"
],
```
Command `expo build:android -t apk` works here, no need to use EAS build.

**Comment by iBryan6** [comment]
The problem is that many people can't downgrade because of the work that was done on top of this SDK and adding "ACCESS_BACKGROUND_LOCATION" to the `app.json` file requires a video and explanation of why we need this...

**Comment by MarouaniALA** [comment]
any news on this please

**Comment by bracketized** [comment]
I think the only hope we have to get this resolved is to wait until SDK44 and pray for it.

**Comment by uzbeki** [comment]
So just to be clear, this would not happen if I were to use bare project with react native without using expo, right?

**Comment by brentvatne** [comment]
@uzbeki - build a development client: https://docs.expo.dev/development/introduction/ 

this was fixed in expo go in https://github.com/expo/expo/pull/15462 - we'll release a new version of expo go for sdk 44 beta soon, and shortly after that we'll release it to the play store.

**Comment by BasselTurky** [comment]
My solution:

I am using Eas build development client.

This is the error I got : 
[Unhandled promise rejection: Error: You need to add `ACCESS_BACKGROUND_LOCATION` to the AndroidManifest.]

What I did:
Go to your project directory, > android folder > app > src > main > AndroidManifest.xml

add ``` <uses-permission
    android:name="android.permission.ACCESS_BACKGROUND_LOCATION" /> ```

just like this: 

```  
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.basselturky.waypointexpo">
<uses-permission
    android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />

  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
  <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
  <uses-permission android:name="android.permission.INTERNET"/>
  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
  <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
  <uses-permission android:name="android.permission.VIBRATE"/>
  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
...
```

I hope it helps, good luck.

**Comment by janusqa** [comment]
I just ran into this issue.

I added the below to expo.plugins in the app.config instead of adding the permission directly under the android node.
```
          "isIosBackgroundLocationEnabled": true,
          "isAndroidBackgroundLocationEnabled": true
```
On local machine there is still no "ACCESS_BACKGROUND_LOCATION" but now there is one in the build logs when I submit it to expo.dev to build and the error has disappeared (at least for now)
so it seems its being added for the dev build remotely.
